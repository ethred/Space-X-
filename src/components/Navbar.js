import React from 'react';
import { NavLink } from 'react-router-dom'; // Import NavLink
import './styles/Navbar.css';
import logo from '../assets/planet.png';

const Navbar = () => (
  <header>
    <nav className="navbar">
      <NavLink to="/" exact className="navbar-logo" activeClassName="active">
        <img
          src={logo}
          alt="Logo"
          width="40"
          height="40"
        />
        <span className="navbar-text">SpaceTravelers HUB</span>
      </NavLink>
      <ul className="nav-links">
        <li className="nav-item">
          <NavLink to="/" exact className="nav-link" activeClassName="active">
            Rockets
          </NavLink>
        </li>
        <li className="nav-item">
          <NavLink to="/missions" className="nav-link" activeClassName="active">
            Missions
          </NavLink>
        </li>
        <li>|</li>
        <li className="nav-item">
          <NavLink to="/mypage" className="nav-link" activeClassName="active">
            My Profile
          </NavLink>
        </li>
      </ul>
    </nav>
  </header>
);

export default Navbar;
