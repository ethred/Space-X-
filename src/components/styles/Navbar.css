.navbar {
  background-color: #001f3f;
  color: white;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px 20px;
}

.nav-link {
  text-decoration: none;
  position: relative;
  color: white;
  font-weight: bold;
  transition: color 0.3s;
}

.nav-link::before {
  content: "";
  position: absolute;
  width: 100%;
  height: 2px;
  bottom: 0;
  left: 0;
  background-color: #ffd700;
  transform: scaleX(0);
  transform-origin: bottom left;
  transition: transform 0.5s ease-in-out;
}

.nav-link.active::before {
  transform: scaleX(1);
}

/* Navbar logo */
.navbar-logo {
  display: flex;
  align-items: center;
  text-decoration: none;
  color: white;
}

/* Navbar text */
.navbar-text {
  font-size: 1.2rem;
  font-weight: bold;
}

/* Navbar links */
.nav-links {
  list-style: none;
  display: flex;
  gap: 20px;
}

.nav-item {
  margin: 0;
}

.nav-link:hover {
  color: #ffd700;
}

.navbar-logo img {
  width: 40px;
  height: 40px;
  animation: rotate 4s infinite linear;
  margin-right: 10px;
}

@keyframes rotate {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}
