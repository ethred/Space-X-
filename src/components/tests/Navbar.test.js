// Navbar.test.js
import React from 'react';
import renderer from 'react-test-renderer'; // Import react-test-renderer
import { BrowserRouter as Router } from 'react-router-dom'; // Import BrowserRouter for routing
import Navbar from '../Navbar'; // Import your Navbar component

test('Navbar snapshot test', () => {
  const tree = renderer
    .create(
      <Router>
        <Navbar />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
