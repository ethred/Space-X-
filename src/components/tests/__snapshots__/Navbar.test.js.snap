// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot test 1`] = `
<header>
  <nav
    className="navbar"
  >
    <a
      activeClassName="active"
      aria-current="page"
      className="navbar-logo active"
      exact={true}
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Logo"
        height="40"
        src="planet.png"
        width="40"
      />
      <span
        className="navbar-text"
      >
        SpaceTravelers HUB
      </span>
    </a>
    <ul
      className="nav-links"
    >
      <li
        className="nav-item"
      >
        <a
          activeClassName="active"
          aria-current="page"
          className="nav-link active"
          exact={true}
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          activeClassName="active"
          className="nav-link"
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li>
        |
      </li>
      <li
        className="nav-item"
      >
        <a
          activeClassName="active"
          className="nav-link"
          href="/mypage"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
